### PR TITLE
README: update instructions and usage guide

### PR DESCRIPTION
**Pull request by lpawelcz**
Here are some initial updates to the README. Some of the changes describe modifications that will come in follow-up PRs with regards to handling of docker images. Please let us know if there are some additional topics that you would want to have included in this README.
There will be also an update to the megaboom README (e.g. pointing to bazel-orfs README for initial setup and usage guide)

CC @oharboe

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -122,25 +138,53 @@ For details about targets spawned by this macro please refer to `Implementation`
...
+Docker flow uses containerized environment with preinstalled ORFS to run the physical design flow.
+Example targets which run the docker flow include:
+
+* //:L1MetadataArray_test_floorplan
+* //:L1MetadataArray_test_memory
+* //:tag_array_64x184_synth
+
+It implicitly depends on a docker image with installed ORFS environment being present in docker runtime of the machine running bazel targets.
+The docker image used in the flow defaults to `ghcr.io/antmicro/openroad-flow-scripts/ubuntu22.04:latest`.
+It can be also overwritten per `build_openroad` instance with a `docker_image` attribute.
```
Comment (oharboe) [inline README.md:168]: "overwritten" is confusing here. "The default can be overridden.."

File: README.md [hunk 2]
```diff
@@ -122,25 +138,53 @@ For details about targets spawned by this macro please refer to `Implementation`
...
+Setting this attribute to a valid registry and image withing this registry will enable docker to automatically pull the image if it's not available locally.
```
Comment (oharboe) [inline README.md:169]: spelling "withing"

File: README.md [hunk 3]
```diff
@@ -122,25 +138,53 @@ For details about targets spawned by this macro please refer to `Implementation`
...
+User can also build the image from ORFS sources following [the guide](https://openroad-flow-scripts.readthedocs.io/en/latest/user/BuildWithDocker.html#build-using-docker-from-sources).
```
Comment (oharboe) [inline README.md:170]: "User" => "Users"

**Review by oharboe — CHANGES_REQUESTED** [review]
nits

File: README.md [hunk 4]
```diff
@@ -122,25 +138,53 @@ For details about targets spawned by this macro please refer to `Implementation`
 This file contains simple helper functions written in starlark as well as macro `build_openroad()`.
 The implementation of this macro spawns multiple `genrule` native rules which are responsible for preparing and running ORFS physical design flow targets during bazel build stage.
 
-There are 5 kinds of genrules spawned in this macro:
+There are 6 kinds of genrules spawned in this macro:
```
[the remark this answers is not in the document]
Comment (lpawelcz) [reply README.md]: fixed

File: README.md (hunk 1 above)
Comment (lpawelcz) [reply README.md]: fixed

File: README.md (hunk 2 above)
Comment (lpawelcz) [reply README.md]: fixed

File: README.md (hunk 3 above)
Comment (lpawelcz) [reply README.md]: fixed

**Comment by lpawelcz** [comment]
Good catches, thank you.
I also rephrased a little bit the paragraph about fetching docker image as this will be the next thing I will improve

**Comment by lpawelcz** [comment]
Bumped after deciding to leave `orfs_env` bazel target in https://github.com/The-OpenROAD-Project/bazel-orfs/pull/25 (for reference: https://github.com/The-OpenROAD-Project/bazel-orfs/pull/22#issuecomment-2058600945)

**Pull request merged**